reference-implementation/to-upstream-wpts/piping/general.js

```diff
@@ -154,4 +155,55 @@ promise_test(() => {
 
 }, 'Piping from a ReadableStream for which a chunk becomes asynchronously readable after the pipeTo');
 
+function duckTypedPassThroughTransform() {
+  let enqueueInReadable;
+  let closeReadable;
+
+  return {
+    writable: new WritableStream({
+      write(chunk) {
+        enqueueInReadable(chunk);
+      },
+
+      close() {
+        closeReadable();
+      }
+    }),
+
+    readable: new ReadableStream({
+      start(c) {
+        enqueueInReadable = c.enqueue.bind(c);
+        closeReadable = c.close.bind(c);
+      }
+    })
+  };
+}
+
+promise_test(() => {
+  const readableEnd = sequentialReadableStream(5).pipeThrough(duckTypedPassThroughTransform());
+
+  return readableStreamToArray(readableEnd).then(chunks =>
+    assert_array_equals(chunks, [1, 2, 3, 4, 5]));
+}, 'Piping through a duck-typed pass-through transform stream should work');
+
+promise_test(() => {
+  const rs = new ReadableStream({
+    start(c) {
+      c.enqueue('a');
+      c.enqueue('b');
+      c.enqueue('c');
+      c.close();
+    }
+  });
+
+  const ts = new TransformStream();
```

Comment:
I don't think we need the `ts` variable. I think it would be clearer just to create the TransformStream object inline.